src/Controller/BackendCsvImportController.php

```diff
@@ -0,0 +1,356 @@
+    {
+        return $this->importFromTemplate(
+            function ($data, $row) {
+                return array_merge($data, $row);
+            },
+            $dc->table,
+            'listitems',
+            $dc->id,
+            $GLOBALS['TL_LANG']['MSC']['lw_import'][0],
+            true
+        );
+    }
+
+    /**
+     * Imports CSV data for table wizard in content element.
+     *
+     * @param DataContainer $dc
+     *
+     * @return Response
+     */
+    public function importTableWizard(DataContainer $dc)
+    {
+        return $this->importFromTemplate(
+            function ($data, $row) {
+                $data[] = $row;
+
+                return $data;
+            },
+            $dc->table,
+            'tableitems',
+            $dc->id,
+            $GLOBALS['TL_LANG']['MSC']['tw_import'][0]
+        );
+    }
+
+    /**
+     * Imports CSV data for form field options.
+     *
+     * @param DataContainer $dc
+     *
+     * @return Response
+     */
+    public function importOptionWizard(DataContainer $dc)
+    {
+        return $this->importFromTemplate(
+            function ($data, $row) {
+                $data[] = [
+                    'value'   => $row[0],
+                    'label'   => $row[1],
+                    'default' => $row[2] ? 1 : '',
+                    'group'   => $row[3] ? 1 : '',
+                ];
+
+                return $data;
+            },
+            $dc->table,
+            'options',
+            $dc->id,
+            $GLOBALS['TL_LANG']['MSC']['ow_import'][0]
+        );
+    }
+
+    /**
+     * Runs the default import routine with a Contao template.
+     *
+     * @param callable    $callback
+     * @param string      $table
+     * @param string      $field
+     * @param int         $id
+     * @param string|null $submitLabel
+     * @param bool        $allowLinebreak
+     *
+     * @return Response
+     */
+    protected function importFromTemplate(
+        callable $callback,
+        $table,
+        $field,
+        $id,
+        $submitLabel = null,
+        $allowLinebreak = false
+    ) {
+        $this->framework->initialize();
+
+        $request  = $this->requestStack->getCurrentRequest();
+        $uploader = new FileUpload();
+        $template = $this->prepareTemplate($request, $uploader, $allowLinebreak);
+
+        if (null !== $submitLabel) {
+            $template->submitLabel = $submitLabel;
+        }
+
+        if ($request->request->get('FORM_SUBMIT') === $this->getFormId($request)) {
+            try {
+                $data = $this->fetchData($uploader, $request->request->get('separator'), $callback);
+            } catch (\RuntimeException $e) {
+                $request->getSession()->getFlashBag()->add($e->getMessage());
+
+                return new RedirectResponse($request->getUri(), 303);
+            }
+
+            $this->connection->update(
+                $table,
+                [$field => serialize($data)],
```

Comment:
This doesn't quite feel right to me. What if my widget stores the data elsewhere? Then I cannot use this method anymore, right? Can we maybe abstract this a little? Maybe using a `callable` like in other methods here?